### PR TITLE
[webgpu-ep] Fix test_batchnorm_example

### PR DESCRIPTION
**Pull request by jchen10**
This fixes the missing component handling for the input and output variables in BatchNorm operator.

### CODE REVIEW

**Comment by jchen10** [comment]
The corresponding [code lines](https://github.com/microsoft/onnxruntime/blob/4d5e274f0e45533227cbc2a2eed2bec83d42949b/js/web/lib/wasm/jsep/webgpu/ops/batch-norm.ts#L66) in js-ep operator.

**Comment by jchen10** [comment]
@fs-eire @guschmue PTAL

**Comment by fs-eire** [comment]
I think the shader cache is missing too: https://github.com/microsoft/onnxruntime/blob/4d5e274f0e45533227cbc2a2eed2bec83d42949b/js/web/lib/wasm/jsep/webgpu/ops/batch-norm.ts#L125

**Review by guschmue — DISMISSED** [review]

**Comment by jchen10** [comment]
> I think the shader cache is missing too:
> 
> https://github.com/microsoft/onnxruntime/blob/4d5e274f0e45533227cbc2a2eed2bec83d42949b/js/web/lib/wasm/jsep/webgpu/ops/batch-norm.ts#L125

Good catch, done!

**Review by fs-eire — APPROVED** [review]

**Review by guschmue — APPROVED** [review]
